common/models/acl.js

```diff
@@ -231,6 +231,16 @@ module.exports = function(ACL) {
       }
       if (!req.isWildcard()) {
         // We should stop from the first match for non-wildcard
+        permission = candidate.permission;
+
+        // Not really. We must find the first permission that matches our request
+        var new_candidate = acls.find((acl) => {
+          return ((acl.principalType === principalType) && (acl.principalId === principalId));
```

Comment:
I tried out this PR as a monkey patch on my system. I had to add this condition:
```
(acl.accessType === req.accessType)
```
otherwise the permissions check would do things like allow write access when only read access was granted.